[ADD] helpdesk: track and bill 15.0 (#8821)
odoo/documentation: content/applications/services/helpdesk/advanced/track_and_bill.rst

   manually saved and refreshed before the :guilabel:`Project` field appears.

The project selected in this field represents where all the timesheets for this team's tickets are
recorded. Click into the :guilabel:`Project` drop-down menu to select a project.

To create a new project where the timesheets are recorded, click into the :guilabel:`Project`
drop-down menu, type a name for the project, and then click :guilabel:`Create` from the drop-down
menu beneath.

.. image:: track_and_bill/track-bill-enable-settings.png
   :align: center
   :alt: The settings page of a Helpdesk team with the track and bill time settings enabled.

.. _helpdesk/configure-service-products:

Configure service products
~~~~~~~~~~~~~~~~~~~~~~~~~~

When the :guilabel:`Time Billing` feature is enabled, a new product is created in the *Sales* app
called *Service on Timesheets*. This product can be found under :menuselection:`Sales app-->
Products --> Products`. Then, search for `Service on Timesheets` in the :guilabel:`Search...` bar.
This is the product that is used when invoicing for *post-paid support services* **after** they have

> been completed.

Reviewer: Is 'post-paid' a term for Odoo, or is it called 'after-market' meaning after purchasing? Just curious- not a necessary change.

Author: I don't know for sure if it is just an Odoo term, but I wanted to use the terms from the previous docs I am basing this one on

> Select :guilabel:`Service on Timesheets` from the product page. This reveals the product detail

Reviewer: ```suggestion
Select :guilabel:`Service on Timesheet` from the product page. This reveals the product detail
```

form. The product is configured with the :guilabel:`Product Type` set to :guilabel:`Service` and the
:guilabel:`Invoicing Policy` set to :guilabel:`Based on Timesheets`. Make any necessary changes to
the product record, such as the :guilabel:`Cost` or :guilabel:`Sales Price`.

.. image:: track_and_bill/track-bill-product-based-on-timesheets.png
   :align: center
   :alt: The Service on Timesheets product with emphasis on the invoicing policy field.

In order to invoice for support services *before* the work has been completed, also known as
*prepaid support services*, a separate product with a different invoicing policy must be created.

To create a new service product, go to :menuselection:`Sales app --> Products --> Products`, and
click :guilabel:`Create`. This reveals a blank product detail form.

On the new product form, add a :guilabel:`Product Name`, and set the :guilabel:`Product Type` to

> :guilabel:`Service`. Then, set the :guilabel:`Invoicing Policy` to :guilabel:`Prepaid/Fixed Price`.

Reviewer: You could add a sample name, like `Prepaid Service` or a similar example.

This means an invoice can be generated and payment can be received for this product before any
timesheets entries have been recorded for these services.

.. image:: track_and_bill/track-bill-product-prepaid-fixed.png
   :align: center
   :alt: The Service on Timesheets product with emphasis on the invoicing policy field.

Finally, set the :guilabel:`Sales Price`, and confirm that the :guilabel:`Unit of Measure` is set to
:guilabel:`Hours`.

Invoice prepaid support services
================================

When support services are billed on a fixed price, an invoice can be created before any work is
completed on the issue. In this case, a service product with the *Invoicing Policy* set to
*Prepaid/Fixed Price* would be used, just like the one created in :ref:`the section above

> <helpdesk/configure-service-products>`.

Reviewer: is this like doc links, where the name of the sectin shoudl be used instead of a generic 'secton above'? I know for docs I was told to do that, so now I list the name of the doc instead of say "this documentation".

Create a sales order with prepaid product

> -----------------------------------------

Reviewer: Suggesting either "a prepaid product' or "prepaid products"

To invoice a customer for prepaid support services, first create a sales order (SO) with the support
services product. To do this, go to :menuselection:`Sales app --> Orders --> Quotations`. Then,
click :guilabel:`Create` to reveal a blank quotation form.

Then, fill out the quotation form with the customer information.

Go to the :guilabel:`Order Lines` tab of the quotation and click :guilabel:`Add a Product`. Then,

> select the *prepaid services product* configured in the steps above. Update the :guilabel:`Quantity`

Reviewer: could link to the section in this line.

field with the number of hours.

After updating any other necessary information, :guilabel:`Confirm` the quotation. This converts the
quotation into an :abbr:`SO (sales order)`.

Create and send an invoice for prepaid services
-----------------------------------------------

Once the :abbr:`SO (sales order)` has been confirmed, click the :guilabel:`Create Invoice` button.
This opens a :guilabel:`Create invoices` pop-up window.

If no down payment is collected, the :guilabel:`Create Invoice` type can remain as

> :guilabel:`Regular Invoice`. If a :doc:`down payment <../../../sales/sales/invoicing/down_payment>`

Reviewer: ```suggestion
:guilabel:`Regular invoice`. If a :doc:`down payment <../../../sales/sales/invoicing/down_payment>`
```

is collected, choose between either :guilabel:`Down payment (percentage)` or :guilabel:`Down payment
(fixed amount)`.

When the necessary information has been entered, click :guilabel:`Create and View Invoice` or
:guilabel:`Create Invoice`.

.. tip::
   Invoices are created in draft mode, so they can be reviewed and edited if necessary.

The invoice can then be :doc:`sent to the customer <../../../finance/accounting/customer_invoices>`
for payment.

> Create helpdesk ticket for prepaid services

Reviewer: ```suggestion
Create a helpdesk ticket for prepaid services
```

or helpdesk tickets

-------------------------------------------

To create a *Helpdesk* ticket for prepaid services, navigate to :menuselection:`Helpdesk` and click
the :guilabel:`Tickets` button to reveal a specific team's pipeline. Click :guilabel:`Create` to
create a new ticket.

> On the blank ticket form, create a ticket :guilabel:`Title`, and enter the :guilabel:`Customer`

Reviewer: ```suggestion
On the blank ticket form, enter a ticket :guilabel:`Title` and the :guilabel:`Customer`
```
create just sounded odd since you type it in- a silly personal suggestion =)

information.

When the customer name is added, the :guilabel:`Sales Order Item` field automatically populates with

> the most recent prepaid sales order item that has time remaining.

Reviewer: the most recent sales order- specifically for service, right? Might want to clarify- it's only or billable service hours as opposed to like, a table or something (if that's true).

Author: "the most recent item that has time remaining"

If a customer has more than one sales order item with remaining time, click the :guilabel:`Sales
Order Item` field and select the correct item from the drop-down list.

After entering all of the necessary information, click :guilabel:`Save`.

Track hours on helpdesk ticket
------------------------------

Time spent working on a *Helpdesk* ticket is tracked on the *Timesheets* tab on the specific ticket.

On the ticket detail form, click on the :guilabel:`Timesheets` tab and click :guilabel:`Add a line`.
Choose an :guilabel:`Employee`, add a :guilabel:`Description` of the task, and enter the number of

> :guilabel:`Hours Spent` working on the task.

Reviewer: Do ou want to mention they can click the Start button and track realtime, live? Or link to a doc explainng that aspect?

As new lines are added to :guilabel:`Timesheets` tab, the :guilabel:`Remaining Hours on SO` field,

> at the bottom-right of the tab, is automatically updated.

Reviewer: In my runbot, I was able to select a timesheet that did notdisplay the remining time. i think it's because there wasn't time left- probably not common, but maybe reference that? I oly disocverd it because I couldn't see the 'Remaining Hours' so I switche the timesheet. I think this is in all runbots if you want to test it. If it's importnat info- could be put into a note.

.. image:: track_and_bill/track-bill-remaining-hours-total.png
   :align: center
   :alt: The timesheets tab of a Helpdesk ticket keeping track of the number of hours remaining on a
       sales order.

.. note::
   If the number of hours on the :guilabel:`Timesheets` tab exceeds the number of hours sold, the
   :guilabel:`Remaining Hours of SO` turns red.

As hours are added to the :guilabel:`Timesheets` tab, they are automatically updated in the
:guilabel:`Delivered` field on the :abbr:`SO (sales order)`, as well.

Invoice post-paid support services
==================================

> When support services are billed based on the amount of time spent on an issue, an invoice cannot be

Reviewer: ```suggestion
When support services are billed based on the amount of time spent on an issue, an invoice **cannot** be
```

created before the total number of hours required to solve the problem have been entered on a
timesheet. In this case, a service product with the *Invoicing Policy* set to *Based on Timesheets*

> would be used, like the one created in :ref:`above <helpdesk/configure-service-products>`.

Reviewer: ```suggestion
would be used, like the one created :ref:`above <helpdesk/configure-service-products>`.
```

Create a sales order with a time-tracked product
------------------------------------------------

To invoice a customer for post-paid support services, first create a sales order (SO) with the
:guilabel:`Service on Timesheets` product. To do this, go to :menuselection:`Sales app --> Orders
--> Quotations`. Then, click :guilabel:`Create` to reveal a blank quotation form.

Fill out the quotation with the customer information.

> On the :guilabel:`Order Lines` tab, click :guilabel:`Add a Product`. Select the :guilabel:`Service

Reviewer: you can make the "service on timesheets" be the ref link nd save a few words (totally optional)

on Timesheets` product configured :ref:`above <helpdesk/configure-service-products>`. After updating
any other necessary information, :guilabel:`Confirm` the quotation.

.. important::

>    Unlike with the prepaid services quotation, *Odoo* does **not** allow an invoice to be created at

Reviewer: Do we italicize Odoo?

>    Unlike with the prepaid services quotation, *Odoo* does **not** allow an invoice to be created at

Reviewer: ```suggestion
   Unlike with the prepaid service quotations, *Odoo* does **not** allow an invoice to be created at
```

   delivered, therefore, there is nothing to invoice.

Create a helpdesk ticket for time-tracked services
--------------------------------------------------

To record a *Timesheet* entry for time-tracked services, go to the :menuselection:`Helpdesk` app,
and select the appropriate team for which these services apply.

If there is already an existing ticket for this issue, select it from the Kanban view to open it. If
there is no existing ticket for this customer issue, click :guilabel:`Create` to create a new ticket
and enter the necessary customer information on the blank ticket details form.

After selecting or creating a ticket, go to the :guilabel:`Sales Order Item` drop-down menu. Select
the :abbr:`SO (sales order)` created in the previous step.

Track support hours on a ticket
-------------------------------

In order to create an invoice for a product based on timesheets, hours need to be tracked and
recorded. At this point, the service is considered *delivered*. To record hours for this support
service, click on the :guilabel:`Timesheets` tab of the ticket.

Click :guilabel:`Add a Line` to record a new entry. Select an :guilabel:`Employee` from the

> drop-down menu, and record the time spent in the :guilabel:`Hours Spent` column.

Reviewer: Would this be a better place to have the start/stop button mentioned?

menu beneath.

.. image:: track_and_bill/track-bill-enable-settings.png
   :align: center
   :alt: The settings page of a Helpdesk team with the track and bill time settings enabled.

.. _helpdesk/configure-service-products:

Configure service products
~~~~~~~~~~~~~~~~~~~~~~~~~~

When the :guilabel:`Time Billing` feature is enabled, a new product is created in the *Sales* app
called *Service on Timesheets*. This product can be found under :menuselection:`Sales app-->
Products --> Products`. Then, search for `Service on Timesheet` in the :guilabel:`Search...` bar.
This is the product that is used when invoicing for *post-paid support services* **after** they have
been completed.

Select :guilabel:`Service on Timesheet` from the product page. This reveals the product detail
form. The product is configured with the :guilabel:`Product Type` set to :guilabel:`Service` and the
:guilabel:`Invoicing Policy` set to :guilabel:`Based on Timesheets`. Make any necessary changes to
the product record, such as the :guilabel:`Cost` or :guilabel:`Sales Price`.

.. image:: track_and_bill/track-bill-product-based-on-timesheets.png
   :align: center

>    :alt: The Service on Timesheets product with emphasis on the invoicing policy field.

Reviewer 2: ```suggestion
   :alt: The Service on Timesheet product with emphasis on the invoicing policy field.
```

In order to invoice for support services *before* the work has been completed, also known as
*prepaid support services*, a separate product with a different invoicing policy must be created.

To create a new service product, go to :menuselection:`Sales app --> Products --> Products`, and
click :guilabel:`Create`. This reveals a blank product detail form.

On the new product form, add a :guilabel:`Product Name`.

.. tip::
   Try to use a name that identifies the purpose of the product, for example `Prepaid Services`.
   This makes it easier when adding it to a sales order later.

Set the :guilabel:`Product Type` to :guilabel:`Service`. Then, set the :guilabel:`Invoicing Policy`
to :guilabel:`Prepaid/Fixed Price`. This means an invoice can be generated and payment can be
received for this product before any timesheets entries have been recorded for these services.

.. image:: track_and_bill/track-bill-product-prepaid-fixed.png
   :align: center

>    :alt: The Service on Timesheets product with emphasis on the invoicing policy field.

Reviewer 2: ```suggestion
   :alt: The Service on Timesheet product with emphasis on the invoicing policy field.
```

:guilabel:`Hours`.

Invoice prepaid support services
================================

When support services are billed on a fixed price, an invoice can be created before any work is
completed on the issue. In this case, a service product with the *Invoicing Policy* set to
*Prepaid/Fixed Price* would be used, just like the one created in :ref:`the section above
<helpdesk/configure-service-products>`.

Create a sales order with a prepaid product
-------------------------------------------

To invoice a customer for prepaid support services, first create a sales order (SO) with the support
services product. To do this, go to :menuselection:`Sales app --> Orders --> Quotations`. Then,
click :guilabel:`Create` to reveal a blank quotation form.

Then, fill out the quotation form with the customer information.

Go to the :guilabel:`Order Lines` tab of the quotation and click :guilabel:`Add a Product`. Then,
select the *prepaid services product* configured in the steps :ref:`above
<helpdesk/configure-service-products>`. Update the :guilabel:`Quantity` field with the number of
hours.

After updating any other necessary information, :guilabel:`Confirm` the quotation. This converts the

> quotation into an :abbr:`SO (sales order)`.

Reviewer 2: ```suggestion
quotation into a :abbr:`SO (sales order)`.
```

Create and send an invoice for prepaid services
-----------------------------------------------

Once the :abbr:`SO (sales order)` has been confirmed, click the :guilabel:`Create Invoice` button.
This opens a :guilabel:`Create invoices` pop-up window.

If no down payment is collected, the :guilabel:`Create Invoice` type can remain as
:guilabel:`Regular invoice`. If a :doc:`down payment <../../../sales/sales/invoicing/down_payment>`
is collected, choose between either :guilabel:`Down payment (percentage)` or :guilabel:`Down payment
(fixed amount)`.

When the necessary information has been entered, click :guilabel:`Create and View Invoice` or
:guilabel:`Create Invoice`.

.. tip::

>    Invoices are created in draft mode, so they can be reviewed and edited if necessary.

Reviewer 2: ```suggestion
   Invoices are created in draft mode, so they can be reviewed and edited, if necessary.
```

The invoice can then be :doc:`sent to the customer <../../../finance/accounting/customer_invoices>`

> for payment.

Reviewer 2: ```suggestion
for payment, via the :guilabel:`Send By Email` button.
```
I believe that's correct, right? If not, politely disregard this suggestion. 👍

> Create a helpdesk ticket for prepaid services

Reviewer 2: ```suggestion
Create helpdesk ticket for prepaid services
```

---------------------------------------------

> To create a *Helpdesk* ticket for prepaid services, navigate to :menuselection:`Helpdesk` and click

Reviewer 2: ```suggestion
To create a *Helpdesk* ticket for prepaid services, navigate to :menuselection:`Helpdesk app`, and click
```

> the :guilabel:`Tickets` button to reveal a specific team's pipeline. Click :guilabel:`Create` to

Reviewer 2: ```suggestion
the :guilabel:`Tickets` button on the desired team's card, in order to reveal that specific team's pipeline. Click :guilabel:`Create` to
```

create a new ticket.

On the blank ticket form, enter a ticket :guilabel:`Title` and the :guilabel:`Customer`
information.

When the customer name is added, the :guilabel:`Sales Order Item` field automatically populates with
the most recent prepaid sales order item that has time remaining.

If a customer has more than one sales order item with remaining time, click the :guilabel:`Sales

> Order Item` field and select the correct item from the drop-down list.

Reviewer 2: ```suggestion
Order Item` field, and select the correct item from the drop-down list.
```

After entering all of the necessary information, click :guilabel:`Save`.

Track hours on helpdesk ticket
------------------------------

Time spent working on a *Helpdesk* ticket is tracked on the *Timesheets* tab on the specific ticket.

> On the ticket detail form, click on the :guilabel:`Timesheets` tab and click :guilabel:`Add a line`.

Reviewer 2: ```suggestion
On the ticket detail form, click on the :guilabel:`Timesheets` tab, and click :guilabel:`Add a line`.
```

:guilabel:`Hours Spent` working on the task.

As new lines are added to :guilabel:`Timesheets` tab, the :guilabel:`Remaining Hours on SO` field,
at the bottom-right of the tab, is automatically updated.

.. image:: track_and_bill/track-bill-remaining-hours-total.png
   :align: center
   :alt: The timesheets tab of a Helpdesk ticket keeping track of the number of hours remaining on a
       sales order.

.. note::
   If the number of hours on the :guilabel:`Timesheets` tab exceeds the number of hours sold, the
   :guilabel:`Remaining Hours of SO` turns red.

As hours are added to the :guilabel:`Timesheets` tab, they are automatically updated in the
:guilabel:`Delivered` field on the :abbr:`SO (sales order)`, as well.

Invoice post-paid support services
==================================

When support services are billed based on the amount of time spent on an issue, an invoice
**cannot** be created before the total number of hours required to solve the problem have been
entered on a timesheet. In this case, a service product with the *Invoicing Policy* set to *Based on
Timesheets* would be used, like the one created :ref:`above <helpdesk/configure-service-products>`.

> Create a sales order with a time-tracked product

Reviewer 2: ```suggestion
Create time-tracked product sales order
```
just an optional suggestion to make this section title a bit more concise

------------------------------------------------

To invoice a customer for post-paid support services, first create a sales order (SO) with the

> :guilabel:`Service on Timesheets` product. To do this, go to :menuselection:`Sales app --> Orders

Reviewer 2: ```suggestion
:guilabel:`Service on Timesheet` product. To do this, go to :menuselection:`Sales app --> Orders
```

--> Quotations`. Then, click :guilabel:`Create` to reveal a blank quotation form.

Fill out the quotation with the customer information.

On the :guilabel:`Order Lines` tab, click :guilabel:`Add a Product`. Select the :guilabel:`Service

> on Timesheets` product configured :ref:`above <helpdesk/configure-service-products>`. After updating

Reviewer 2: ```suggestion
on Timesheet` product configured :ref:`above <helpdesk/configure-service-products>`. After updating
```

any other necessary information, :guilabel:`Confirm` the quotation.

.. important::

>    Unlike with the prepaid service quotations, *Odoo* does **not** allow an invoice to be created at

Reviewer 2: ```suggestion
   Unlike with the prepaid service quotations, Odoo does **not** allow an invoice to be created at
```

   this time. That is because no services have been performed; in other words, nothing has been

>    delivered, therefore, there is nothing to invoice.

Reviewer 2: I am actually still seeing a 'Create Invoice' button after confirming a quotation with just the 'Service on Timesheet' product in the Order Lines tab. Your note makes *total* sense, so it might be worth a quick double-check about this admonition during your revisions.

Create a helpdesk ticket for time-tracked services
--------------------------------------------------

> To record a *Timesheet* entry for time-tracked services, go to the :menuselection:`Helpdesk` app,

Reviewer 2: ```suggestion
To record a *Timesheet* entry for time-tracked services, go to the :menuselection:`Helpdesk app`,
```

and select the appropriate team for which these services apply.

If there is already an existing ticket for this issue, select it from the Kanban view to open it. If

> there is no existing ticket for this customer issue, click :guilabel:`Create` to create a new ticket

Reviewer 2: ```suggestion
there is no existing ticket for this customer issue, click :guilabel:`Create` to create a new ticket,
```

> and enter the necessary customer information on the blank ticket details form.

Reviewer 2: ```suggestion
and enter the necessary customer information on the blank ticket detail form.
```

After selecting or creating a ticket, go to the :guilabel:`Sales Order Item` drop-down menu. Select
the :abbr:`SO (sales order)` created in the previous step.

Track support hours on a ticket
-------------------------------

In order to create an invoice for a product based on timesheets, hours need to be tracked and
recorded. At this point, the service is considered *delivered*. To record hours for this support
service, click on the :guilabel:`Timesheets` tab of the ticket.

Click :guilabel:`Add a Line` to record a new entry. Select an :guilabel:`Employee` from the
drop-down menu, and record the time spent in the :guilabel:`Hours Spent` column.

> Repeat these steps as needed until all time spent on the issues has been recorded.

Reviewer 2: ```suggestion
Repeat these steps, as needed, until all time spent on the issue(s) have been recorded. Then, click :guilabel:`Save`.
```

> Create an invoice for hours tracked on a ticket

Reviewer 2: ```suggestion
Create invoice for hours tracked on a ticket
```

and to bill a customer for that time. Through integrations with the *Sales*, *Timesheets*,
*Project*, and *Accounting* applications, customers can be charged once the work is completed, or
before it has even begun.

.. warning::
   Since the *Track & Bill Time* features require integration with other applications, enabling them
   may result in the installation of additional modules or applications.

   Installing a new application on a *One-App-Free* database triggers a 15-day trial. At the end of
   the trial, if a paid subscription has not been added to the database, it will no longer be active
   or accessible.

Track and bill time features configuration
==========================================

Before a customer can be invoiced for support services, the *Track & Bill Time* features **must** be
enabled on each *Helpdesk* team individually.

Enable track and bill time on a team
------------------------------------

To view and enable the *Track & Bill Time* features on a *Helpdesk* team, first navigate to
:menuselection:`Helpdesk app --> Configuration --> Helpdesk Teams`. Then, select a team from the

> list, or create a :doc:`new one <../overview/getting_started>`. This reveals a team's settings page.

Reviewer 3: This is a nitpick 😄 please use meaningful/contextual label names for links. This helps specify exactly which section you are referring to as well as show the reader where the link will take them.
```suggestion
list, or create a :doc:`new Helpdesk team <../overview/getting_started>`. This reveals a team's
settings page.
```

click :guilabel:`Create`. This reveals a blank product detail form.

On the new product form, add a :guilabel:`Product Name`.

.. tip::
   Try to use a name that identifies the purpose of the product, for example, `Prepaid Services`.
   This makes it easier when adding it to a sales order later.

Set the :guilabel:`Product Type` to :guilabel:`Service`. Then, set the :guilabel:`Invoicing Policy`
to :guilabel:`Prepaid/Fixed Price`. This means an invoice can be generated and payment can be
received for this product, before any timesheet entries have been recorded for these services.

.. image:: track_and_bill/track-bill-product-prepaid-fixed.png
   :align: center
   :alt: The Service on Timesheet product with emphasis on the invoicing policy field.

Finally, set the :guilabel:`Sales Price`, and confirm that the :guilabel:`Unit of Measure` is set to
:guilabel:`Hours`. Then, click :guilabel:`Save` to save those configurations.

Invoice prepaid support services
================================

When support services are billed on a fixed price, an invoice can be created before any work is
completed on the issue. In this case, a service product with the *Invoicing Policy* set to

> *Prepaid/Fixed Price* would be used, just like the one created in :ref:`the section above
> <helpdesk/configure-service-products>`.

Reviewer 3: ```suggestion
*Prepaid/Fixed Price* would be used, just like the one created in the :ref:`Configure service
products <helpdesk/configure-service-products>` section.
```
